src/api/common/HttpWrapper.ts

```diff
@@ -0,0 +1,81 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse,
+} from 'axios';
+
+const LOGMAXLEN = 120;
+const DEFAULT_TIMEOUT = 30000;
+
+const stringify = (something: unknown): string =>
+  JSON.stringify(something, null, 3);
+
+const interceptors = {
+  logRequest: {
+    onResponse: (response: AxiosResponse) => {
+      console.log(
+        `Got a response: ${response.config.url as string}${stringify(
+          response.data
+        ).slice(0, LOGMAXLEN)}`
+      );
+
+      return response;
+    },
+    onError: (error: AxiosError) => {
+      const { response, message, config } = error;
+      if (error.request) {
+        console.log(`Request failed: ${stringify(error.request)}`);
+      }
+      if (response) {
+        console.log(`Response error: ${stringify(response)}`);
+      }
+      if (!error.request && !response) {
+        console.log('Error while setting up request', stringify(message));
+      }
+      console.log(`Configuration: ${stringify(config)}`);
+
+      return Promise.reject(error);
+    },
+  },
+};
+
+export default class HttpWrapper {
+  private readonly _httpClient: AxiosInstance;
+
+  private readonly _isLogEnabled: boolean;
+
+  constructor(opts: AxiosRequestConfig = {}, isLogEnabled = false) {
+    this._httpClient = axios.create({
+      timeout: DEFAULT_TIMEOUT,
+      headers: { 'Content-Type': 'application/json' },
+      ...opts,
+    });
+    this._isLogEnabled = isLogEnabled;
+
+    if (this._isLogEnabled) {
+      this._httpClient.interceptors.response.use(
+        interceptors.logRequest.onResponse,
+        interceptors.logRequest.onError
+      );
+    }
+  }
+
+  async sendPromise(url: string, jsonRequestData?: unknown): Promise<unknown> {
```

Comment:
This could probably be a generic function 
```suggestion
  async sendPromise<T>(url: string, jsonRequestData?: unknown): Promise<T> {
```

Reply:
fair. will change

Reply:
done